src/win/i2c.cc

```diff
@@ -0,0 +1,318 @@
+#include "i2c.h"
+#include <ppltasks.h>
+#include <vector>
+#include <iostream>
+
+using v8::FunctionTemplate;
+using namespace concurrency;
+using namespace Platform;
+using namespace Windows::Devices::Enumeration;
+
+Nan::Persistent<v8::Function> WinI2c::constructor;
+
+NAN_MODULE_INIT(WinI2c::Init) {
+  v8::Local<v8::FunctionTemplate> tpl = Nan::New<v8::FunctionTemplate>(New);
+  tpl->SetClassName(Nan::New("WinI2c").ToLocalChecked());
+  tpl->InstanceTemplate()->SetInternalFieldCount(6);
+
+  Nan::SetPrototypeMethod(tpl, "openSync", OpenSync);
+  Nan::SetPrototypeMethod(tpl, "closeSync", CloseSync);
+  Nan::SetPrototypeMethod(tpl, "read", Read);
+  Nan::SetPrototypeMethod(tpl, "readPartial", ReadPartial);
+  Nan::SetPrototypeMethod(tpl, "write", Write);
+  Nan::SetPrototypeMethod(tpl, "writePartial", WritePartial);
+  Nan::SetPrototypeMethod(tpl, "writeRead", WriteRead);
+  Nan::SetPrototypeMethod(tpl, "writeReadPartial", WriteReadPartial);
+  Nan::SetPrototypeMethod(tpl, "setDevice", SetDevice);
+
+  constructor.Reset(Nan::GetFunction(tpl).ToLocalChecked());
+  Nan::Set(target, Nan::New("WinI2c").ToLocalChecked(), Nan::GetFunction(tpl).ToLocalChecked());
+}
+
+WinI2c::WinI2c() {
+  RoInitialize(RO_INIT_MULTITHREADED);
+}
+
+WinI2c::~WinI2c() {
+  RoUninitialize();
+}
+
+NAN_METHOD(WinI2c::New) {
+  if (info.IsConstructCall()) {
+    WinI2c *obj = new WinI2c();
+    obj->Wrap(info.This());
+    info.GetReturnValue().Set(info.This());
+  }
+  else {
+    const int argc = 1;
+    v8::Local<v8::Value> argv[argc] = { info[0] };
+    v8::Local<v8::Function> cons = Nan::New(constructor);
+    info.GetReturnValue().Set(cons->NewInstance(argc, argv));
+  }
+}
+
+NAN_METHOD(WinI2c::OpenSync) {
+  if (info.Length() < 1 ||
+      !info[0]->IsString()) {
+    return Nan::ThrowError(Nan::ErrnoException(EINVAL, "WinI2c::OpenSync",
+      "incorrect arguments passed to WinI2c::OpenSync"
+      "(string i2cControllerName)"));
+  }
+
+  Nan::HandleScope scope;
```

Comment:
An implicit HandleScope is created on JavaScript accessible methods. Why is another one created here?